src/common/transformations/tests/op_conversions/convert_gather_to_compressed_test.cpp

```diff
@@ -225,3 +225,47 @@ TEST_F(TransformationTestsF, ConvertGatherToCompressedMultiOutput) {
         model_ref = std::make_shared<ov::Model>(ov::NodeVector{gather_compressed}, ov::ParameterVector{input1, input2});
     }
 }
+
+TEST_F(TransformationTestsF, ConvertGatherToCompressedFP16Weight) {
+    {
+        auto input1 = std::make_shared<ov::op::v0::Parameter>(ov::element::i32, ov::PartialShape{-1, 16});
+        auto axis_const = ov::op::v0::Constant::create(ov::element::i32, ov::Shape{1}, {1});
+        auto weights_const = ov::op::v0::Constant::create(ov::element::f16, ov::Shape{32, 16}, {1});
+        auto convert = std::make_shared<ov::op::v0::Convert>(weights_const, ov::element::f32);
+        auto gather = std::make_shared<ov::op::v8::Gather>(convert, input1, axis_const);
+
+        model = std::make_shared<ov::Model>(ov::NodeVector{gather}, ov::ParameterVector{input1});
+        manager.register_pass<ConvertGatherToGatherCompressed>();
+    }
+    {
+        auto input1 = std::make_shared<ov::op::v0::Parameter>(ov::element::i32, ov::PartialShape{-1, 16});
+        auto axis_const = ov::op::v0::Constant::create(ov::element::i32, ov::Shape{1}, {1});
+        auto weights_const = ov::op::v0::Constant::create(ov::element::f16, ov::Shape{32, 16}, {1});
+        auto gather = std::make_shared<ov::op::v8::Gather>(weights_const, input1, axis_const);
```

Comment:
Shouldn't `GatherCompressed` be here?

Reply:
`GatherCompressed` node is to support compressed weight with scales(+shift), such as i4/u4/nf4/u8, which decompression weight inside `GatherCompressed`. Compressed weight with f16/bf16 will not generate `GatherCompressed` node, it is still a gather node, just move convert(f16/bf16 to f32) after gather.
I add explanation comment on pass head file, maybe it helpful to understand.

Reply:
BTW, I did not change case 1 part:https://github.com/openvinotoolkit/openvino/blob/4d1b03a8ffbc1b284faa982f6b275a0bd62e8111/src/common/transformations/src/transformations/op_conversions/convert_gather_to_compressed.cpp#L71-L155

Reply:
1. `Gathercompressed` is executed without jit kernel and `static_cast` to any output precisions. `Gather` create and use jit kernel conditionally. So need update all `gather `code pass to support mix precision.
2. Mix precision(real16 input and f32 output) should not always be true, as if node after gather still accept real16 precision.  Should modify pass or at least verify to ensure the precision propagation work as expect.
@dmitry-gorokhov Mixed precision support is risk to include 25.1. I plan do it in separate PR, what do you think?
cc @maxnick.

Reply:
Thanks for the detailed explanation. Since the main goal (memory consumption reduction) is already achieved, I don't mind to implement mixed precision support in follow up PR.

Reply:
@chenhu-wang Yeah, I think this is reasonable to proceed step by step. 
Lets just add detailed comment that describes the reason why we create Gather instead of GatherCompressed